llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -519,10 +519,12 @@ MipsTargetLowering::MipsTargetLowering(const MipsTargetMachine &TM,
   setTargetDAGCombine({ISD::SDIVREM, ISD::UDIVREM, ISD::SELECT, ISD::AND,
                        ISD::OR, ISD::ADD, ISD::SUB, ISD::AssertZext, ISD::SHL});
 
-  if (Subtarget.isGP64bit())
+  if (ABI.IsN32() || ABI.IsN64())
```

Comment:
Any difference between `Subtarget.isGP64bit()` and `ABI.IsN32() || ABI.IsN64()`.
If we don't support O64 ABI, I guess that they are the same?
And in fact if we'd support O64, `Subtarget.isGP64bit()` may be better.

Reply:
And if we do something like
```
clang -mabi=32 -march=mips64r2
```
may be we wish set it to 64 here.